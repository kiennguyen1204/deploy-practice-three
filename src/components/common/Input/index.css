.text-input-default {
  display: flex;
  width: var(--width-base);
  padding: 4px 4px 4px 24px;
  align-items: center;
  width: var(--width-base);
  gap: 8px;
  border-radius: var(--border-radius-huge);
  border: 1px solid var(--color-matte-white);
  background: var(--color-neutral-white);
}

.text-wrapper {
  width: var(--width-base);
}

.text-input.input-sm {
  padding: 4px 4px 4px 24px;
}

.text-input.input-md {
  padding: 24px 44px;
}
