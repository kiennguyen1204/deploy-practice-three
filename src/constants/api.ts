export const URL_API = {
  BASE_URL: process.env.VITE_BASE_URL,
  PRODUCTS_URL: '/products'
};

const PRODUCTS_API = `${URL_API.BASE_URL}${URL_API.PRODUCTS_URL}`;

export { PRODUCTS_API };
