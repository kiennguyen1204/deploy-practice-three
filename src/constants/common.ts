export const MAX_STAR = 5;
