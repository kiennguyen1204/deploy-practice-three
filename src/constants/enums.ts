export enum BUTTON_VARIANT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary'
}

export enum BUTTON_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
  EXTRA_LARGE = 'xl'
}

export enum BUTTON_COLOR {
  SUCCESS = 'success',
  WARNING = 'warning',
  DEFAULT = 'default'
}

export enum BUTTON_COLOR_TEXT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary'
}

export enum BUTTON_TYPE {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset'
}

export enum INPUT_VARIANT {
  DEFAULT = 'default',
  PRIMARY = 'primary'
}

export enum INPUT_TYPE {
  TEXT = 'text',
  NUMBER = 'number'
}

export enum INPUT_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
  EXTRA_LARGE = 'xl'
}

export enum STATUS {
  SUCCESS = 'success',
  ERROR = 'error'
}

export enum TYPE_LINKS {
  QUICK = 'quick',
  MORE = 'more',
  PAYMENT = 'payment'
}

export enum OPTION_WEED {
  FLOWERS,
  MUSHROOMS,
  CONCENTRACE
}

export enum STEPS_CART {
  SHOPPING_CART,
  CHECKOUT,
  ORDER_COMPLETE
}

export enum SORT_TYPE {
  ASC = 'price_asc',
  DESC = 'price_desc'
}