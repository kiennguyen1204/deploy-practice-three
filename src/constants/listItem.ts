export const LIST_ITEM_NAVIGATION = [
  { name: 'Shop All', to: '/about' },
  {
    name: 'Flower',
    to: ''
  },
  { name: 'Edibles', to: '' },
  {
    name: 'Concentrates',
    to: ''
  },
  { name: 'Mushrooms', to: '' },
  {
    name: 'Promotions/Bundles',
    to: ''
  },
  {
    name: 'Support',
    to: ''
  },
  { name: 'Rewards', to: '' },
  { name: 'Blog', to: '' }
];

export const DATA_INFO = [
  {
    title: 'Indica',
    image: '/src/assets/images/indica.png',
    icon: '/src/assets/images/sleeping_face.svg',
    description: `This type of cannabis is commonly taken by those who want to sink into a state of total relaxation in every limb. This reduces stress overall and can take your worries and fatigue away. Because of its relaxing effects, it is suggested to use this type of cannabis at night. It is particularly recommended for people who have trouble sleeping, be it due to insomnia, pain or other associated sleep issues.`,
    to: '/indica',
    shop: 'Shop Indica'
  },
  {
    title: 'Sativa',
    image: '/src/assets/images/indica.png',
    icon: '/src/assets/images/sun.svg',
    description:
      'Contrary to the deep all-body relaxation that comes with the use of indica strains, sativas are known for increasing the feeling of creativity, enhancing focus and lessening anxiety. This is more of a mind-centered high in terms of how and where you will feel the effects. Given its stimulating nature, it is recommended to use this during the day.',
    to: '/sativa',
    shop: 'Shop Sativa'
  },
  {
    title: 'Hybrids',
    image: '/src/assets/images/hybrids.png',
    description: `Hybrid strains are just that – they combine the head-focused high effects of sativas with the bodily relaxation of indicas. This is ideal for people who really want to sooth any fatigue and worries while also clearing the mind ready to focus fresh on something new. Hybrids are expertly tailored to result in strains with specific effects. Useful strain vocabulary to familiarize yourself with when it comes to hybrids includes sativa-dominant (full, bodily relaxation), indica-dominant (boosting creativity, increasing mood and lessening anxiety) and balanced (the best of both worlds).`,
    top: '/hybrids',
    shop: 'Shop Hybrids'
  }
];

export const LIST_ITEM_EDUCATION = [
  {
    image: '/src/assets/images/education/info.webp',
    title: '12 Mistakes To Avoid When Buying Cannabis Online',
    subtitle: 'January 24, 2023',
    description:
      'Buying cannabis online can be a great way to get your hands on the products you need without leaving the comfort of your home. But …',
    to: '/home'
  },
  {
    image: '/src/assets/images/education/weight.webp',
    title: 'How To Store Cannabis and Keep it Fresh and Potent?',
    subtitle: 'January 20, 2023',
    description:
      'Cannabis packaging has advanced dramatically in recent years, whether your state has a medicinal marijuana programme, legal adult-use weed, or both. Most ...',
    to: '/home'
  },
  {
    image: '/src/assets/images/education/weight.webp',
    title: 'The Ultimate Guide to Checking the Quality of Cannabis – 10 Industry Leading Tips',
    subtitle: 'January 19, 2023',
    description:
      'Quality cannabis is a term used to describe cannabis products that meet specific standards of excellence. It is essential to understand what quality cannabis means, …',
    to: '/home'
  }
];
