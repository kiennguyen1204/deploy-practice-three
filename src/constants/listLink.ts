import Mastercard from 'assets/images/master_card.svg';
import Visa from 'assets/images/visa.svg';
import Bitcoin from 'assets/images/bitcoin.svg';
import Interac from 'assets/images/interac.svg';

export const QUICK_LINKS = [
  {
    name: 'Track Your Order',
    to: '/home'
  },
  {
    name: 'Shop All',
    to: '/home'
  },
  {
    name: 'Flower',
    to: '/home'
  },
  {
    name: 'Edibles',
    to: '/home'
  },
  {
    name: 'Concentrates',
    to: '/home'
  },
  {
    name: 'Refunds',
    to: '/home'
  },
  {
    name: 'Mushrooms',
    to: '/home'
  },
  {
    name: 'Promotions / Bundles',
    to: '/home'
  },
  {
    name: 'Promotions / Bundles',
    to: '/home'
  },
  {
    name: 'Support',
    to: '/home'
  },
  {
    name: 'Reward',
    to: '/home'
  },
  {
    name: 'Blog',
    to: '/home'
  },
  {
    name: 'Shipping Faq',
    to: '/home'
  }
];

export const MORE_LINKS = [
  {
    name: 'Buy weed online in Canada',
    to: '/'
  },
  {
    name: 'Buy weed online in New Brunswick',
    to: '/'
  },
  {
    name: 'Buy weed online in Prince Edward Island',
    to: '/'
  },
  {
    name: 'Buy weed online in Northwest Territories',
    to: '/'
  },
  {
    name: 'Buy weed online in Saskatchewan',
    to: '/'
  },
  {
    name: 'Buy weed online in Manitoba',
    to: '/'
  },
  {
    name: 'Buy weed online in Quitebec',
    to: '/'
  },
  {
    name: 'Buy weed online in British Columbia',
    to: '/'
  },
  {
    name: 'Buy weed online in Ontario',
    to: '/'
  },
  {
    name: 'Buy weed online in Alberta',
    to: '/'
  }
];

export const PAYMENT_METHODS = [
  {
    name: 'mastercard',
    image: Mastercard
  },
  {
    name: 'visa',
    image: Visa
  },
  {
    name: 'bitcoin',
    image: Bitcoin
  },
  {
    name: 'interac',
    image: Interac
  }
];
