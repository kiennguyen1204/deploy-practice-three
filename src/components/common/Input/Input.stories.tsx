import type { Meta, StoryObj } from '@storybook/react';

// constants
import { INPUT_VARIANT } from 'constants/enums';

// components
import Input from 'components/common/Input';

const meta: Meta<typeof Input> = {
  component: Input
};

type Story = StoryObj<typeof Input>;

export const Primary: Story = {
  args: {
    title: 'Input',
    variant: INPUT_VARIANT.PRIMARY
  }
};

export const Default: Story = {
  args: {
    title: 'Input',
    variant: INPUT_VARIANT.DEFAULT
  }
};

export default meta;
