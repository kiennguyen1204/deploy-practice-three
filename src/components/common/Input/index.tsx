import { ReactElement, memo } from 'react';

// constants
import { INPUT_SIZE, INPUT_TYPE, INPUT_VARIANT } from 'constants/enums';

// styles
import './index.css';

export interface Props {
  name?: string;
  value?: string;
  className?: string;
  placeholder?: string;
  title?: string;
  variant?: INPUT_VARIANT;
  type?: INPUT_TYPE;
  size?: INPUT_SIZE;
  onChange?: (value: string) => void;
}

const Input = ({
  name,
  value,
  placeholder,
  title,
  className,
  variant = INPUT_VARIANT.DEFAULT,
  type = INPUT_TYPE.TEXT,
  size = INPUT_SIZE.MEDIUM,
  onChange
}: Props): ReactElement => {
  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    if (onChange) {
      onChange(newValue);
    }
  };

  return (
    <section className='text-wrapper'>
      {title && <label>{title}</label>}
      <input
        className={`text-input text-input-${variant} input-${size} ${className}`}
        type={type}
        name={name}
        value={value}
        placeholder={placeholder}
        onChange={handleInputChange}
      />
    </section>
  );
};

export default memo(Input);
