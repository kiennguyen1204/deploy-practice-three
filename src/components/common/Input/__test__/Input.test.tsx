import { render, screen, fireEvent } from '@testing-library/react';

// constants
import { INPUT_SIZE, INPUT_TYPE, INPUT_VARIANT } from 'constants/enums';

// components
import Input, { Props } from 'components/common/Input';

const defaultProps: Props = {
  name: 'test-input',
  value: '',
  onChange: jest.fn()
};

const renderInput = (props?: Partial<Props>) => {
  const mergedProps = { ...defaultProps, ...props };
  return render(<Input {...mergedProps} />);
};

describe('Input component', () => {
  it('renders an input field with default props', () => {
    renderInput();

    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('type', 'text');
    expect(inputElement).toHaveAttribute('name', 'test-input');
    expect(inputElement).toHaveValue('');
  });

  it('renders an input field with a custom type', () => {
    renderInput({ type: INPUT_TYPE.TEXT });

    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toHaveAttribute('type', 'text');
  });

  it('renders an input field with a custom placeholder', () => {
    renderInput({ placeholder: 'Enter your name' });

    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toHaveAttribute('placeholder', 'Enter your name');
  });

  it('renders an input field with a custom title (label)', () => {
    renderInput({ title: 'Name' });

    const labelElement = screen.getByText('Name');
    expect(labelElement).toBeInTheDocument();
    expect(labelElement.tagName).toBe('LABEL');
  });

  it('renders an input field with a custom variant class', () => {
    renderInput({ variant: INPUT_VARIANT.PRIMARY });

    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toHaveClass('text-input-primary');
  });

  it('renders an input field with a custom size class', () => {
    renderInput({ size: INPUT_SIZE.LARGE });

    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toHaveClass('input-lg');
  });

  it('calls onChange when input value changes', () => {
    const handleChange = jest.fn();
    renderInput({ onChange: handleChange });

    const inputElement = screen.getByRole('textbox');

    fireEvent.change(inputElement, { target: { value: 'New Value' } });

    expect(handleChange).toHaveBeenCalledWith('New Value');
  });

  it('calls onBlur when input loses focus', () => {
    renderInput();
    const inputElement = screen.getByRole('textbox');
    const handleBlur = jest.fn();

    inputElement.addEventListener('blur', handleBlur);
    fireEvent.blur(inputElement);

    expect(handleBlur).toHaveBeenCalledTimes(1);
  });

  it('calls onKeyDown when a key is pressed in the input', () => {
    renderInput();
    const inputElement = screen.getByRole('textbox');
    const handleKeyDown = jest.fn();

    inputElement.addEventListener('keydown', handleKeyDown);
    fireEvent.keyDown(inputElement, { key: 'Enter' });

    expect(handleKeyDown).toHaveBeenCalledWith(expect.objectContaining({ key: 'Enter' }));
  });
});
