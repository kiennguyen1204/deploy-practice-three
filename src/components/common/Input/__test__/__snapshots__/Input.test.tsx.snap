// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="text-wrapper"
  >
    <input
      class="text-input text-input-default input-md undefined"
      type="text"
      value=""
    />
  </section>
</DocumentFragment>
`;
